contracts/vaults/RibbonThetaVault.sol

```diff
@@ -459,16 +526,53 @@ contract RibbonThetaVault is DSMath, GnosisAuction, OptionsVaultStorage {
         uint256 shortAmount = wmul(freeBalance, lockedRatio);
         lockedAmount = shortAmount;
 
-        GammaProtocol.createShort(
+        GAMMA_PROTOCOL.createShort(
             GAMMA_CONTROLLER,
             MARGIN_POOL,
             newOption,
             shortAmount
         );
 
+        startAuction();
+
         emit OpenShort(newOption, shortAmount, msg.sender);
     }
 
+    /**
+     * @notice Initiate the gnosis auction.
+     */
+    function startAuction() public onlyManager {
```

Comment:
Is there a reason this function needs to be `public`?

Reply:
yes. 

It is not internal because there should be the optionality to redo the auction for the **current** options for whatever unforseen circumstances may arise. For example, there could be some issue where we don't think we have enough oTokens sold (we can adjust this in initiateAuction to not sell at all if we do not reach a goal), and we might want to redo it again for some reason. There are other wild cases that I might not be thinking of that might be worth being ready for at the expense of 250 -> 500 gas. Keeping it internal means it can only be called once for the current option.

it is not external because it is called internally, so it is public

Reply:
Cool, this design of making the auction redo-able is interesting and potentially useful. My feedback on this,

* We should make the `startAuction` function permission-less (remove onlyManager) because it just kicks off the options process. (I don't see a reason why we need it to be permissioned)
* Can we get better guarantees around how often an auction will be kicked off? Let's say we start an auction of 100 otokens, we only sell off 80 otokens. We want to burn the remaining 20 otokens. But when we post the tx to burn the otokens, we get frontrun by bots that call `startAuction` again. I'm not sure what the solution here is, just putting out the possibility of a DoS attack where we are not able to ever burn the otokens.

Reply:
👍